packages/mdc-menu-surface/index.ts

```diff
@@ -146,74 +122,43 @@ class MDCMenuSurface extends MDCComponent {
   }
 
-  /** @param {boolean} quickOpen */
-  set quickOpen(quickOpen) {
-    this.foundation_.setQuickOpen(quickOpen);
-  }
+  getDefaultFoundation(): MDCMenuSurfaceFoundation {
+    // TODO(acdvorak): Should we change the type of root_ to HTMLElement? What other kind of element could it be?
+    // SVGElement appears to have the same methods/properties (at least the ones we care about).
+    const rootEl = this.root_ as HTMLElement;
 
-  /** @return {!MDCMenuSurfaceFoundation} */
-  getDefaultFoundation() {
-    return new MDCMenuSurfaceFoundation(
-      /** @type {!MDCMenuSurfaceAdapter} */ (Object.assign({
-        addClass: (className) => this.root_.classList.add(className),
-        removeClass: (className) => this.root_.classList.remove(className),
-        hasClass: (className) => this.root_.classList.contains(className),
-        hasAnchor: () => !!this.anchorElement,
-        notifyClose: () => this.emit(MDCMenuSurfaceFoundation.strings.CLOSED_EVENT, {}),
-        notifyOpen: () => this.emit(MDCMenuSurfaceFoundation.strings.OPENED_EVENT, {}),
-        isElementInContainer: (el) => this.root_ === el || this.root_.contains(el),
-        isRtl: () => getComputedStyle(this.root_).getPropertyValue('direction') === 'rtl',
-        setTransformOrigin: (origin) => {
-          this.root_.style[`${util.getTransformPropertyName(window)}-origin`] = origin;
-        },
+    return new MDCMenuSurfaceFoundation({
+      addClass: (className: string) => this.root_.classList.add(className),
+      removeClass: (className: string) => this.root_.classList.remove(className),
+      hasClass: (className: string) => this.root_.classList.contains(className),
+      hasAnchor: () => !!this.anchorElement,
+      notifyClose: () => this.emit(MDCMenuSurfaceFoundation.strings.CLOSED_EVENT, {}),
+      notifyOpen: () => this.emit(MDCMenuSurfaceFoundation.strings.OPENED_EVENT, {}),
+      isElementInContainer: (el: Element) => this.root_ === el || this.root_.contains(el),
+      isRtl: () => getComputedStyle(this.root_).getPropertyValue('direction') === 'rtl',
+      setTransformOrigin: (origin: string) => {
+        const propertyName = `${util.getTransformPropertyName(window)}-origin`;
+        rootEl.style.setProperty(propertyName, origin);
       },
-      this.getFocusAdapterMethods_(),
-      this.getDimensionAdapterMethods_())
-      ));
-  }
 
-  /**
-   * @return {!{
-   * isFocused: function(): boolean,
-   * saveFocus: function(),
-   * restoreFocus: function(),
-   * isFirstElementFocused: function(): boolean,
-   * isLastElementFocused: function(): boolean,
-   * focusFirstElement: function(),
-   * focusLastElement: function(),
-   * }}
-   * @private
-   */
-  getFocusAdapterMethods_() {
-    return {
       isFocused: () => document.activeElement === this.root_,
       saveFocus: () => {
-        this.previousFocus_ = document.activeElement;
+        this.previousFocus_ = document.activeElement as HTMLElement;
```

Comment:
what if `previousFocus` is a union. `Element | HTMLElement | SVGElement`? If that's the case, then I think firstFocusableElement/lastFocustableElement can also be that type.

Reply:
Done